docs/data-migration.md

```diff
@@ -0,0 +1,76 @@
+
+
+## Prepare the destination database
+
+### Using postgres as the destination remote database
+
+Prepare a user and database as described in the [Postgres](postgres.md)
+documentation. You'll need the Data Source Name (DSN) for both the data
+migration and then the `lnd` configuration, so keep that string somewhere
+(should be something with the format of `postgres://xx:yy@localhost:5432/zz`).
+
+No additional steps are required to prepare the Postgres database for the data
+migration. The migration tool will create the database schema automatically, so
+no DDL scripts need to be run in advance.
+
+## Prepare the source database
+
+Assuming we want to migrate the database state from the pre-0.19.0 individual
+`bbolt` based `*.db` files to a remote database, we first need to make sure the
+source files are in the correct state.
+
+The following steps should be performed *before* running the data migration:
+1. Stop `lnd`
+2. Upgrade the `lnd` binary to the latest version (e.g. `v0.19.0-beta` or later)
+3. Make sure to add config options like `gc-canceled-invoices-on-startup=true`
+   and `db.bolt.auto-compact=true` to your `lnd.conf` to optimize the source
+   database size by removing canceled invoices and compacting it on startup.
+4. Start `lnd` normally, using the flags mentioned above but not yet changing
+   any database backend related configuration options. Check the log that the
+   database schema was migrated successfully, for example: `Checking for
+   schema update: latest_version=XX, db_version=XX`
+5. Remove any data from the source database that you can. The fewer entries are
+   in the source database, the quicker the migration will complete. For example
+   failed payments (or their failed HTLC attempts) can be removed with
+   `lncli deletepayments`.
+6. Stop `lnd` again and make sure it isn't started again by accident during the
+   data migration (e.g. disable any `systemd` or other scripts that start/stop
+   `lnd`).
+
+## Run the migration
+
+Depending on the destination database type, run the migration with a command
+similar to one of the following examples:
+
+**Example: Migrate from `bbolt` to `sqlite`:**
+
+```shell
+⛰ lndinit --debuglevel info  migrate-db --source.bolt.   data-dir /home/myuser/.lnd/data  --dest.backend sqlite  --dest.sqlite.data-dir  /home/myuser/.lnd/data  --network mainnet 
+```
+If you were running a watchtower server, and it had a different directory set compared to the default `LND` directory make sure you also add the tower dir setting. It has to be the directory to the `watchtower` dir, excluding the `watchtower` name e.g.:
+`--source.bolt.tower-dir /home/myuser/towerdir`.
+
+**Example: Migrate from `bbolt` to `postgres`:**
+
+```shell
+⛰  lndinit -v migrate-db \
```

Comment:
this `-v` flag doesnt seem to be recognized and causes it to error out:

```
2025-03-28 10:15:20.797 [INF]: LNDINIT Version 0.1.22-beta commit=, debuglevel=
2025-03-28 10:15:20.797 [ERR]: LNDINIT Config error: unknown flag `v'
```

Reply:
yes that was a previous logging setting, it now changed to `--debuglevel`